src/utils/checkDeviceHasWebcam.ts

```diff
@@ -0,0 +1,45 @@
+export async function checkDeviceHasWebcam(): Promise<boolean> {
+  let hasWebcam = false;
+
+  // If we cannot enumerate devices
+  if (
+    !(typeof MediaStreamTrack !== 'undefined' && 'getSources' in MediaStreamTrack)
+      && !navigator.mediaDevices?.enumerateDevices
+  ) {
+    return false;
+  }
+
+  if (navigator.mediaDevices?.enumerateDevices) {
+    navigator.enumerateDevices = async (callback) => {
+      await navigator.mediaDevices.enumerateDevices()
+        ?.then(callback)
+        ?.catch(() => {
+          callback([]);
+        });
+    };
+  }
+
+  if (
+    !navigator.enumerateDevices
+      && window.MediaStreamTrack
+      && (window.MediaStreamTrack as any).getSources
```

Comment:
Wrong indentation and probably can be written in shorter way.
```suggestion
    && (window.MediaStreamTrack as any)?.getSources
```